### PR TITLE
Prevent Notebooks tab from getting stuck refreshing clusters (no JIRA)

### PR DESCRIPTION
**Pull request by MattPutnam**
When you have a cluster in a pending state, the Notebooks tab refreshes clusters every 10 seconds. If one is stuck in this state, it will continue refreshing every 10 seconds, even if you nav away. This prevents that.

Also some formatting that my editor did automatically as I worked.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [x] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Review by davidangb — APPROVED** [review]
:+1: did not test.

File: src/cljs/main/broadfcui/page/workspace/notebooks/notebooks.cljs [hunk 1]
```diff
@@ -311,44 +311,47 @@
...
+                                                                       :data-test-id "create-modal-button"
+                                                                       :onClick #(swap! state assoc :show-create-dialog? true)}]]
+                          :clusters clusters
+                          :reload-after-delete #(this :-get-clusters-list-if-whitelisted))]))]))
    :component-did-mount
    (fn [{:keys [this]}]
      (this :-is-leo-whitelisted))
+   :component-will-unmount
+   (fn [{:keys [locals]}]
+     (swap! locals assoc :dead? true))
```
Comment (aednichols) [inline src/cljs/main/broadfcui/page/workspace/notebooks/notebooks.cljs:333]: I thought unmounting took place immediately before a component is destroyed. How does the value of `:dead?` have an effect in this state?

**Review by aednichols — COMMENTED** [review]
This looks reasonable, but I would like to test it out.

I tried a few hacks to `:-is-leo-whitelisted` so notebooks would appear, but I couldn't get it to work.

**Review by aednichols — APPROVED** [review]
Got access to notebooks in dev, tested & looks good

👍

File: src/cljs/main/broadfcui/page/workspace/notebooks/notebooks.cljs (hunk 1 above)
Comment (MattPutnam) [reply src/cljs/main/broadfcui/page/workspace/notebooks/notebooks.cljs:333]: The component is unmounted, but the JS hangs around.

**Comment by MattPutnam** [comment]
jenkins retest

**Comment by MattPutnam** [comment]
Error is unrelated.

**Comment by rtitle** [comment]
Cool, this looks good to me. Thanks for fixing.

I have a [PR](https://github.com/broadinstitute/firecloud-ui/pull/1111) which adds some more AJAX calls for Running clusters. If this merges first I can update my code to check the `:dead?` flag as well.

If you have a few mins, would you mind reviewing it? https://github.com/broadinstitute/firecloud-ui/pull/1111

**Pull request merged**